src/main/java/io/jenkins/plugins/artifact_manager_jclouds/s3/S3BlobStoreConfig.java

```diff
@@ -133,6 +148,30 @@ public void setRegion(String region) {
         save();
     }
 
+    public String getCredentialsId() {
+        return credentialsId;
+    }
+
+    @DataBoundSetter
+    public void setCredentialsId(String credentialsId) {
+        this.credentialsId = credentialsId;
+        save();
+    }
+
+    public AWSCredentialsImpl getCredentials() {
+        Jenkins.get().checkPermission(Jenkins.ADMINISTER);
+        Optional<AmazonWebServicesCredentials> credential = CredentialsProvider.lookupCredentials(
+                AmazonWebServicesCredentials.class, Jenkins.get(),ACL.SYSTEM, Collections.emptyList())
+                                                                               .stream()
+                                                                               .filter(it -> it.getId().equals(credentialsId))
+                                                                               .findFirst();
+        if(credential.isPresent() && credential.get() instanceof AWSCredentialsImpl){
```

Comment:
I think you want to refer to `AmazonWebServicesCredentials`, never the implementation class.

Reply:
sadly no, I want to be sure that I grab a AWSCredentialsImpl class, the interface AmazonWebServicesCredentials does not expose the getCredentials() method that I need.

https://github.com/jenkinsci/aws-credentials-plugin/blob/master/src/main/java/com/cloudbees/jenkins/plugins/awscredentials/AWSCredentialsImpl.java#L111

Reply:
> AmazonWebServicesCredentials does not expose the getCredentials() method

Yes it does. It is inherited from the `com.amazonaws.auth.AWSCredentialsProvider` interface.

Reply:
done